### PR TITLE
scope: allow same-name options from different scopes

### PR DESCRIPTION
**Pull request by MarcelCoding**

### CODE REVIEW

File: libixx/src/index.rs [hunk 1]
```diff
@@ -157,7 +157,15 @@ impl Index {
         .options
         .iter()
         .enumerate()
-        .find(|(idx, OptionEntry { labels: option, .. })| do_labels_match(*idx, option, &labels))
+        .find(
+          |(
+            idx,
+            OptionEntry {
+              scope_id: option_scope_id,
```
Comment (SuperSandro2000) [inline libixx/src/index.rs:164]: Are we now searching through all scopes at once or one scope at the time?

File: libixx/src/test/mod.rs [hunk 2]
```diff
@@ -14,55 +14,70 @@ fn test() {
...
       .search(Some(1), "enablenixpkgsreleasecheck", 10)
       .unwrap(),
     vec![]
   );
+
+  // TEST options with same name in different scopes
+  assert_eq!(
+    index.search(None, "ho*debug", 10).unwrap(),
+    vec![
+      (0, 0, "home.enableDebugInfo".to_string()),
```
Comment (SuperSandro2000) [inline libixx/src/test/mod.rs:69]:
```suggestion
      (0, 0, "home.enableDebugInfo".to_string()),
      (0, 1, "home.enable".to_string()),
```
Just to also have a result which is never returned

File: libixx/src/test/mod.rs [hunk 3]
```diff
@@ -14,55 +14,70 @@ fn test() {
...
+      (9, 1, "home.enableDebugInfo".to_string())
+    ]
+  );
+
+  assert_eq!(
```
Comment (SuperSandro2000) [inline libixx/src/test/mod.rs:74]: Maybe we should also test a none match?

**Comment by SuperSandro2000** [comment]
What happens if we are in all scopes and we search for an exact option name which exists in multiple scopes?

File: libixx/src/index.rs (hunk 1 above)
Comment (MarcelCoding) [reply libixx/src/index.rs:164]: This is not the search function. This functions determinate the index of the option in the json metadata. Because option names are now only unique within a scope, the scope id must be included in the parameters of this query function to unique identify the option.

File: libixx/src/test/mod.rs (hunk 2 above)
Comment (MarcelCoding) [reply libixx/src/test/mod.rs:69]: What do you mean here? This assertion would fail with that change.

File: libixx/src/test/mod.rs (hunk 3 above)
Comment (MarcelCoding) [reply libixx/src/test/mod.rs:74]: What do you mean "non match"? We compare the vector. That means that additional elements would also fail the assertion.

**Comment by MarcelCoding** [comment]
The search result appears multiple times in the option list and if clicked on it, it would display the according scope.

File: libixx/src/test/mod.rs (hunk 2 above)
Comment (SuperSandro2000) [reply libixx/src/test/mod.rs:69]: Are we re-using the list of options from above? Didn't see that on the smartphone

Comment (MarcelCoding) [reply libixx/src/test/mod.rs:69]: Yeah I use the same set of options for all assertions.

**Comment by SuperSandro2000** [comment]
We should probably add a label or so to search but that's in the other PR.
